Source/Particles/Deposition/CurrentDeposition.H

```diff
@@ -98,6 +103,16 @@ void doDepositionShapeN(const GetParticlePosition& GetPosition,
     amrex::ParallelFor(
         np_to_depose,
         [=] AMREX_GPU_DEVICE (long ip) {
+#if (defined AMREX_USE_GPU)
+            // To store present thread's time for deposition
+            __shared__ amrex::Real cost_shared;
```

Comment:
offline discussed: this might need to become a global device memory variable for SYCL devices (`AMREX_USE_DPCPP`).
All three: CUDA, HIP, SYCL are positive on `AMREX_USE_GPU`.

Reply:
I think this is okay for CUDA and HIP, but as @WeiqunZhang explained to me shared memory works differently for DPC++.  For now I am leaving the DPC++ case to have each thread add its time to `cost` instead of first accumulating in shared memory.  This makes sense to me for now, since I haven't done any profiling of the DPC++ version.